### PR TITLE
[6CORE-025] added REST Specification for Warehouse Service: Inventory Exploration

### PR DESCRIPTION
**Pull request by YuraKhilko**
Created REST Specification for Warehouse Service: Inventory Exploration

### CODE REVIEW

File: 3-platform/warehouse/spec/src/main/java/com/_6core/platform/warehouse/rest/v1/InventoryGetApi.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        description = "REST specification for getting inventory")
+public interface InventoryGetApi {
+
+    /**
+     * Getting inventory by id GET /inventories/{id}
+     *
+     * @param id - required
+     * @return OK (status code 200)
+     */
+    @GetMapping(value = "/inventories/{id}", produces = "application/json")
```
Comment (sokimaaa) [inline 3-platform/warehouse/spec/src/main/java/com/_6core/platform/warehouse/rest/v1/InventoryGetApi.java]: inventories are chained to warehouses. 
product are chained to inventories. 

change URI, e.g. `/warehouser/{...}/inventories/{...}`

File: 3-platform/warehouse/spec/src/main/java/com/_6core/platform/warehouse/rest/v1/inventory/Inventory.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package com._6core.platform.warehouse.rest.v1.inventory;
+
+public record Inventory(
```
Comment (sokimaaa) [inline 3-platform/warehouse/spec/src/main/java/com/_6core/platform/warehouse/rest/v1/inventory/Inventory.java]: Is it response? request? pls clarify. 
Stay consistent in naming of dto objects.

```suggestion
public record Inventory(
```

File: 3-platform/warehouse/spec/src/main/java/com/_6core/platform/warehouse/rest/v1/inventory/Inventory.java [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+        String inventoryId,
+        String warehouseId,
```
Comment (sokimaaa) [inline 3-platform/warehouse/spec/src/main/java/com/_6core/platform/warehouse/rest/v1/inventory/Inventory.java]: we dont need to return warehouseId, mercant should know where is he store items. 
```suggestion
```

File: 3-platform/warehouse/spec/src/main/java/com/_6core/platform/warehouse/rest/v1/inventory/Inventory.java [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+        String productId,
+        Integer actualQuantity,
+        Integer availableQuantity
```
Comment (sokimaaa) [inline 3-platform/warehouse/spec/src/main/java/com/_6core/platform/warehouse/rest/v1/inventory/Inventory.java]: actual and available quantities are our system implementation.
user should know nothing about the system implementation. 
```suggestion
        Integer quantity
```

File: 3-platform/warehouse/spec/src/main/java/com/_6core/platform/warehouse/rest/v1/InventoryGetApi.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+        Inventory mockInventory = new Inventory(
+                "2_7",
+                "2",
+                "7",
+                10,
+                3
+        );
+        return Flux.just(ResponseEntity
+                .status(HttpStatusCode.valueOf(200))
+                .body(mockInventory));
```
Comment (sokimaaa) [inline 3-platform/warehouse/spec/src/main/java/com/_6core/platform/warehouse/rest/v1/InventoryGetApi.java]:
```suggestion
        return Flux.just(ResponseEntity.ok(mock));
```

File: 3-platform/warehouse/spec/src/main/java/com/_6core/platform/warehouse/rest/v1/InventoryGetApi.java [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+     *
+     * @param id - required
+     * @return OK (status code 200)
+     */
+    @GetMapping(value = "/inventories/{id}", produces = "application/json")
+    @Operation(summary = "Get inventory by id", description = "Get inventory by id")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "OK", content = {
+                    @Content(mediaType = "application/json", schema = @Schema(implementation = Inventory.class))}),
+            @ApiResponse(responseCode = "401", description = "Authentication error"),
```
Comment (sokimaaa) [inline 3-platform/warehouse/spec/src/main/java/com/_6core/platform/warehouse/rest/v1/InventoryGetApi.java:31]: what about 404?

**Review by sokimaaa — CHANGES_REQUESTED** [review]

File: 3-platform/warehouse/spec/src/main/java/com/_6core/platform/warehouse/rest/v1/InventoryGetApi.java (hunk 1 above)
Comment (YuraKhilko) [reply 3-platform/warehouse/spec/src/main/java/com/_6core/platform/warehouse/rest/v1/InventoryGetApi.java]: Will it be necessary to search by warehouse? Why do we need to define warehouse in than URI?

File: 3-platform/warehouse/spec/src/main/java/com/_6core/platform/warehouse/rest/v1/inventory/Inventory.java (hunk 2 above)
Comment (YuraKhilko) [reply 3-platform/warehouse/spec/src/main/java/com/_6core/platform/warehouse/rest/v1/inventory/Inventory.java]: It's response. Thanks.

File: 3-platform/warehouse/spec/src/main/java/com/_6core/platform/warehouse/rest/v1/inventory/Inventory.java (hunk 3 above)
Comment (YuraKhilko) [reply 3-platform/warehouse/spec/src/main/java/com/_6core/platform/warehouse/rest/v1/inventory/Inventory.java]: Ok, thanks.

File: 3-platform/warehouse/spec/src/main/java/com/_6core/platform/warehouse/rest/v1/InventoryGetApi.java (hunk 6 above)
Comment (YuraKhilko) [reply 3-platform/warehouse/spec/src/main/java/com/_6core/platform/warehouse/rest/v1/InventoryGetApi.java:31]: Do we have to say, that there is no such record?

Comment (sokimaaa) [reply 3-platform/warehouse/spec/src/main/java/com/_6core/platform/warehouse/rest/v1/InventoryGetApi.java:31]: yes, it is expected that frontend may ask us about non existing inventory or blocked, for example

that should not be visible

File: 3-platform/warehouse/spec/src/main/java/com/_6core/platform/warehouse/rest/v1/InventoryGetApi.java [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+            @ApiResponse(responseCode = "500", description = "Internal server error")})
+    default Flux<ResponseEntity<Inventory>> getInventoryById(@PathVariable String id) {
+        Inventory mockInventory = new Inventory(
```
Comment (kateryna-mykh) [inline 3-platform/warehouse/spec/src/main/java/com/_6core/platform/warehouse/rest/v1/InventoryGetApi.java:34]: Initialization there isn't a builder chain and there are short parameters. These are all can be written in 1 line.

File: 3-platform/warehouse/spec/src/main/java/com/_6core/platform/warehouse/rest/v1/InventoryGetApi.java [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+                "2_7",
+                "2",
+                "7",
+                10,
+                3
+        );
+        return Flux.just(ResponseEntity
+                .status(HttpStatusCode.valueOf(200))
```
Comment (kateryna-mykh) [inline 3-platform/warehouse/spec/src/main/java/com/_6core/platform/warehouse/rest/v1/InventoryGetApi.java]: ResponseEntity.ok().build()

File: 3-platform/warehouse/spec/src/main/java/com/_6core/platform/warehouse/rest/v1/InventoryGetApi.java (hunk 1 above)
Comment (sokimaaa) [reply 3-platform/warehouse/spec/src/main/java/com/_6core/platform/warehouse/rest/v1/InventoryGetApi.java]: inventories belong to specific warehouse, not exist in a vacuum.

File: 3-platform/warehouse/spec/src/main/java/com/_6core/platform/warehouse/rest/v1/InventoryGetApi.java (hunk 8 above)
Comment (YuraKhilko) [reply 3-platform/warehouse/spec/src/main/java/com/_6core/platform/warehouse/rest/v1/InventoryGetApi.java]: that has been already fiexed.

**Review by kateryna-mykh — APPROVED** [review]

**Review by sokimaaa — APPROVED** [review]

**Pull request merged**